### ISSUE TITLE
Multi-GPU Not Working - Only One GPU Used Despite `device=[...]` Configuration

### ISSUE THREAD

**Issue by anishabhatnagar** [lightonai/fast-plaid#10]
## Summary:
When building a FastPlaid index with device=[f"cuda:{i}" for i in range(torch.cuda.device_count())], only cuda:0 shows utilization in nvidia-smi. Index builds successfully but there’s no apparent multi-GPU scaling. 

## Minimal repro

```python
import torch
from fast_plaid import search

if __name__ == "__main__":
    # Device setup in main block for multiprocessing
    if torch.cuda.is_available():
        device = [f"cuda:{i}" for i in range(torch.cuda.device_count())] 
    else:
        device = "cpu"
    print(f"Using device: {device}")
    
    # FastPlaid initialization in main block
    fastplaid = search.FastPlaid(index="test_index", device=device)
    
    # Generate and process batches
    is_first = True
    for batch_num in range(3):
        print(f"Processing batch {batch_num + 1}")
        
        # Generate dumy documents with varying token lengths
        batch = []
        for doc in range(10000):
            num_tokens = torch.randint(50, 300, (1,)).item()
            doc_embedding = torch.randn(num_tokens, 128)
            batch.append(doc_embedding)
        
        if is_first:
            print("Creating index...")
            fastplaid.create(documents_embeddings=batch)
            is_first = False
        else:
            print("Updating index...")
            fastplaid.update(documents_embeddings=batch)
    
```

## What I See
- console output : ` Using device: ['cuda:0', 'cuda:1', 'cuda:2', 'cuda:3', 'cuda:4', 'cuda:5', 'cuda:6', 'cuda:7']`
- `nvidia-smi`: Only GPU 0 active, others at 0% utilization
- Index builds successfully but no multi-GPU scaling

## Environment

- 2×48GB NVIDIA GPUs
- Large batches (100k+ vectors, 128-dim)
- All GPUs visible before execution

## Questions

1. Are there logging flags or environment variables to trace device allocation and process creation?
2. Configuration needed? Are there additional parameters beyond the device list to enable multi-GPU functionality? The README suggests that multiprocessing will be automatic after setting device list.

I am working on a tight deadline, and would appreciate any help on this!
